test-suite failes with 8.2001.0

Version: 8.2001.0
Debian sid

Trying to build the latest version of rsyslog for Debian, I get

```
PASS: immark.sh
... [SNIPPED by admin for easier readability]
SKIP: imfile-logrotate-nocopytruncate.sh
FAIL: imfile-growing-file-id.sh
PASS: glbl-oversizeMsg-truncate-imfile.sh
PASS: config_enabled-on.sh
PASS: config_enabled-off.sh
PASS: imfile-endmsg.regex-with-example.sh
make[6]: Entering directory '/home/test/rsyslog-8.2001.0/tests'
make[6]: Nothing to be done for 'imfile-escapelf.replacement.log'.
make[6]: Nothing to be done for 'imfile-escapelf.replacement-empty.log'.
make[6]: Leaving directory '/home/test/rsyslog-8.2001.0/tests'
fatal: making test-suite.log: failed to create imfile-escapelf.replacement.trs
fatal: making test-suite.log: failed to create imfile-escapelf.replacement.log
fatal: making test-suite.log: failed to create imfile-escapelf.replacement-empty.trs
fatal: making test-suite.log: failed to create imfile-escapelf.replacement-empty.log
make[5]: *** [Makefile:3414: test-suite.log] Error 1
```

--- a/plugins/imfile/imfile.c
+++ b/plugins/imfile/imfile.c
@@ -1258,8 +1258,8 @@ get_file_id_hash(const char *data, size_t lendata,
 static void ATTR_NONNULL(1)
 getFileID(act_obj_t *const act)
 {
-	/* save the old id for cleaning purposes */
-	strncpy(act->file_id_prev, (const char*)act->file_id, FILE_ID_HASH_SIZE);
+	char tmp_id[FILE_ID_HASH_SIZE];
+	strncpy(tmp_id, (const char*)act->file_id, FILE_ID_HASH_SIZE);
 	act->file_id[0] = '\0';
 	assert(act->fd >= 0); /* fd must have been opened at act_obj_t creation! */
 	char filedata[FILE_ID_SIZE];
@@ -1269,6 +1269,9 @@ getFileID(act_obj_t *const act)
 		get_file_id_hash(filedata, sizeof(filedata), act->file_id, sizeof(act->file_id));
 	} else {
 		DBGPRINTF("getFileID partial or error read, ret %d\n", r);
+	}
+	if (strncmp(tmp_id, act->file_id, FILE_ID_HASH_SIZE)) {/* save the old id for cleaning purposes */
+		strncpy(act->file_id_prev, tmp_id, FILE_ID_HASH_SIZE);
 	}
 	DBGPRINTF("getFileID for '%s', file_id_hash '%s'\n", act->name, act->file_id);
 }

